### PR TITLE
openPMD plugin: Set default backend depending on available options

### PR DESCRIPTION
**Pull request by franzpoeschel**
If only HDF5 is available, we should automatically use that as a backend.

### CODE REVIEW

File: include/picongpu/plugins/openPMD/openPMDWriter.hpp [hunk 1]
```diff
@@ -496,7 +510,7 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
 
                 HDINLINE void operator()(ThreadParams* params)
                 {
-#ifndef __CUDA_ARCH__
+#    ifndef __CUDA_ARCH__
```
Comment (franzpoeschel) [inline include/picongpu/plugins/openPMD/openPMDWriter.hpp]: clang-format is being weird with those

**Review by sbastrakov — DISMISSED** [review]

File: include/picongpu/plugins/openPMD/openPMDWriter.hpp (hunk 1 above)
Comment (sbastrakov) [reply include/picongpu/plugins/openPMD/openPMDWriter.hpp]: There is a clang-format issue in CI (still?). Could it be that you are using a different version from our CI and so this part gets treated differently.

**Comment by psychocoderHPC** [comment]
@franzpoeschel  Please run clang format to fix the formation issues.

File: include/picongpu/plugins/openPMD/openPMDWriter.hpp (hunk 1 above)
Comment (franzpoeschel) [reply include/picongpu/plugins/openPMD/openPMDWriter.hpp]: clang-format on my machine was in a weird mood yesterday, now it's doing this correctly again

**Comment by franzpoeschel** [comment]
@psychocoderHPC done
seems like the CI timeouted?

**Review by steindev — APPROVED** [review]

**Pull request merged**